### PR TITLE
Destination postgres: DV2 beta implementation

### PR DESCRIPTION
**Pull request by edgao**
copied from https://github.com/airbytehq/airbyte/pull/33910, with a merge from master to hopefully fix tests. Removes the spec option from postgres, since we're not quite ready for beta release yet.

test syncs:
* v1 https://cloud.airbyte.com/workspaces/b61bc266-ef3c-460c-af2f-f70da4a2993c/connections/c2f4da3d-f01b-4010-9cb0-08ed0a5fbbdd/job-history#7427892::0
    * running with normalization, no errors yet. expected to complete in a few more hours
* v2 https://cloud.airbyte.com/workspaces/b61bc266-ef3c-460c-af2f-f70da4a2993c/connections/b0b82b70-a732-4aad-b623-13d5c5a75de2/job-history#7412903::0
    * completed successfully using v2
    * the spec option is gone, but I configured this previously to run with v2 - the config option is just invisible now 😅 

assuming these both run correctly overnight, that should be sufficient evidence that syncs are still functioning correctly.

### CODE REVIEW

File: airbyte-cdk/java/airbyte-cdk/core/src/main/java/io/airbyte/cdk/db/factory/DataSourceFactory.java [hunk 1]
```diff
@@ -272,6 +278,8 @@ public DataSource build() {
        */
       config.setInitializationFailTimeout(Integer.MIN_VALUE);
 
+      config.setConnectionInitSql(connectionInitSql);
```
Comment (gisripa) [inline airbyte-cdk/java/airbyte-cdk/core/src/main/java/io/airbyte/cdk/db/factory/DataSourceFactory.java:281]: I should probably remove the `RedshiftDestinationHandler#execute` and test with this to see SET var works consistently

File: airbyte-integrations/connectors/destination-postgres/src/test-integration/java/io/airbyte/integrations/destination/postgres/typing_deduping/PostgresSqlGeneratorIntegrationTest.java [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+
+  /**
+   * See
+   * {@link io.airbyte.integrations.destination.redshift.typing_deduping.RedshiftSqlGeneratorIntegrationTest.RedshiftSourceOperations}.
+   * Copied here to avoid weird dependencies.
+   */
+  public static class PostgresSourceOperations extends JdbcSourceOperations {
+
+    @Override
+    public void copyToJsonField(final ResultSet resultSet, final int colIndex, final ObjectNode json) throws SQLException {
```
Comment (gisripa) [inline airbyte-integrations/connectors/destination-postgres/src/test-integration/java/io/airbyte/integrations/destination/postgres/typing_deduping/PostgresSqlGeneratorIntegrationTest.java:54]: Nice, just one override and looks like TZ related stuff worked out of the box from JdbcSourceOperations

**Comment by gisripa** [comment]
Looks good to me. Can you rebase (sorry there will be a bump of conflicts from my dependency cleanup) and publish CDK since the tests are failing to find 0.13.0 artifact

**Comment by edgao** [comment]
thanks for looking! I'll do that post-long-weekend... feels irresponsible to publish artifacts from a 1k line pr right now :P

though for the record - the last "real" commit was https://github.com/airbytehq/airbyte/pull/34177/commits/4084e63dd6f6729b95abba5ed5445d8cd0e00dc0, where the failures are all because of the version bump check. The [diff](https://github.com/airbytehq/airbyte/pull/34177/files/4084e63dd6f6729b95abba5ed5445d8cd0e00dc0..b6ed088627d20fba1d1bb8f24aafad486842a837) after that is basically just metadata/changelogs (plus removing the `dumb change for cache bust` that we were using to investigate CI).

**Comment by edgao** [comment]
republishing b/c another cdk change got merged to master

**Review by gisripa — APPROVED** [review]

**Pull request merged**